catalog/app/constants/routes.ts

```diff
@@ -0,0 +1,306 @@
+import { mkSearch } from 'utils/NamedRoutes'
+import { encode } from 'utils/s3paths'
+
+const PACKAGE_PATTERN = '[^/]+/[^/]+'
+
+// TODO: make sure types are explicitly divide codebase into
+//       main catalog and embed,
+//       so catalog routes aren't called in embed
+
+type Route<Args extends any[]> = {
+  path: string
+  url: (...args: Args) => string
+}
+
+export type HomeArgs = [options?: { q?: string }]
+
+export const home: Route<HomeArgs> = {
+  path: '/',
+  url: ({ q } = {}) => `/${mkSearch({ q })}`,
+}
+
+export type NoArgs = []
+
+export const install: Route<NoArgs> = {
+  path: '/install',
+  url: () => '/install',
+}
+
+// Marketing
+
+export const about: Route<NoArgs> = {
+  path: '/about',
+  url: () => '/about',
+}
+
+export const personas: Route<NoArgs> = {
+  path: '/personas',
+  url: () => '/personas',
+}
+
+export const product: Route<NoArgs> = {
+  path: '/product',
+  url: () => '/product',
+}
+
+export type ActivateArgs = [options: { registryUrl: string; token: string }]
+
+export const activate: Route<ActivateArgs> = {
```

Comment:
We, probably, have to use some helper, when we migrate to `react-router-dom@v6` anyway. I didn't dig into it completely, but the current routes constants don't just work because `v6` handles nested routes differently

Reply:
> handles nested routes differently

it can handle absolute routes just as well, afaict.
the problem is v6 doesn't support regexes and has some issues with trailing slashes, so it won't be a seamless upgrade, and a helper won't fix it -- we would have to adjust the routes themselves